### PR TITLE
Bj pundit authorization

### PR DESCRIPTION
**Pull request by borvux**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,12 +1,24 @@
 class ApplicationController < ActionController::Base
+  include Pundit
   before_action :authenticate_user!
-
   before_action :configure_permitted_parameters, if: :devise_controller?
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  after_action :verify_authorized, unless: :devise_controller?
```
Comment (ajmompr) [inline app/controllers/application_controller.rb:7]: good job accounting for devise

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class CommentsController < ApplicationController
   before_action :set_comment, only: %i[ show edit update destroy ]
   before_action :is_an_authorized_user, only: [:destroy, :create]
+  before_action { authorize(@comment || Comment) }
```
Comment (ajmompr) [inline app/controllers/comments_controller.rb:4]: good

File: app/controllers/follow_requests_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class FollowRequestsController < ApplicationController
   before_action :set_follow_request, only: %i[ show edit update destroy ]
+  before_action { authorize(@follow_request || FollowRequest) }
```
Comment (ajmompr) [inline app/controllers/follow_requests_controller.rb:3]: good

File: app/controllers/photos_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class PhotosController < ApplicationController
   before_action :set_photo, only: %i[ show edit update destroy ]
   before_action :ensure_current_user_is_owner, only: [:destroy, :update, :edit]
+  before_action { authorize(@photo || Photo) }
```
Comment (ajmompr) [inline app/controllers/photos_controller.rb:4]: good

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action { authorize(@user || User) }
```
Comment (ajmompr) [inline app/controllers/users_controller.rb:3]: good

File: app/policies/user_policy.rb
Comment (ajmompr) [inline app/policies/user_policy.rb:1]: show? should be set to true. Move the logic in show to a new show_photo? method to account for displaying the profiles of private vs public feeds

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show? %>
```
Comment (ajmompr) [inline app/views/users/show.html.erb]: call your new show_photo? policy method here

**Review by ajmompr — CHANGES_REQUESTED** [review]
Fix user policy

**Review by reherr — APPROVED** [review]
looks good to me now!

File: app/views/users/show.html.erb [hunk 7]
```diff
@@ -4,7 +4,7 @@
   </div>
 </div>
 
-<% if current_user == @user || !@user.private? || current_user.leaders.include?(@user) %>
+<% if policy(@user).show_photos? %>
```
Comment (ajmompr) [inline app/views/users/show.html.erb:7]: good

**Review by ajmompr — APPROVED** [review]

**Pull request merged**